src/util/configure.ml

```diff
@@ -20,8 +20,43 @@ let args = [
   "-use-camlp4", arg_bool use_camlp4, " when true enable camlp4 syntax extension";
 ]
 
+let oasis_files = [
+  "setup-dev.exe";
+  "setup.exe";
+  "setup.data";
+  "setup.log";
+  "src/unix/lwt_config.h";
+  "src/unix/lwt_config.ml";
+  "src/unix/lwt_unix_jobs_generated.ml";
+]
+
+let oasis_warning_message = 
+"WARNING!!!
+
+There appear to be some files left behind by the old oasis 
+build system.  This may cause the build to fail.  Please clean 
+the repository before building.  
+
+Deleting the files src/unix/lwt_config.ml and src/unix/lwt_config.h 
+should allow the build to work.
+
+To clean up all oasis generated files you can checkout the old
+build system and run distclean ie
+
+$ git checkout 30c3fd86f9ddebc98e54276100a59c79436b174f
+$ make clean distclean
+$ git checkout master 
+$ git clean -ffd
+"
+
+let oasis_sanity_check () = 
+  if List.fold_left (fun b f -> b || (Sys.file_exists f)) false oasis_files then
```

Comment:
Nit, but I believe this is better implemented by [`List.exists`](https://caml.inria.fr/pub/docs/manual-ocaml/libref/List.html#VALexists). This doesn't block the PR (it doesn't really matter), so I'll merge it in some hours, but if you'd actually like to address this, go ahead :p